### ISSUE TITLE
Google Analytics.

### ISSUE THREAD

**Issue by meetinghouse** [meetinghouse/cms#90]
Andy,
Omitted to say we need Google Analytics. Is this something already present and, if not, is it possible to make it "native", or do we have to install one at a time?
John C.

**Comment by acavanagh** [comment]
You can't make it completely global, but you can make it easy.  Best way to do this is probably to install the tracking code but have it use a variable for the google id,  and then put an admin field in settings where you can enter the ID on a per site basis.  That way you can set a site up for analytics without have to change any code.

**Comment by meetinghouse** [comment]
So what I'm hearing you say is that you would need two things from me for each site:
G Analytics Tracking ID
and
User access level to be granted to the Google ID that is entered through the user interface on the site.

Is that correct?

Is that right?

**Comment by meetinghouse** [comment]
I am trying to test this in preservationarts.corbettresearchgroup.com

I added this client's main site (which is www.preservationarts.net) to John's Google Analytics account and then pasted the Tracking ID into the field in the light theme. So I suppose the code is in there.

I don't have a way to check because I realize now that it isn't really seamless. There's a big old seam in between the two different URL's. The .corbetttresearchgroup URL is intended to be a sort of staging site to put all the content in, and then when it all looks good, we deploy preservationarts.net in Forge and have it basically show whatever is in the preservationarts.corbettresearchgroup.com.

That's how I think it works, anyway. In theory at least, is this going to work for the Google Analytics tracking ID being entered in the light theme field?

**Comment by acavanagh** [comment]
I'm not really an seo guy, but I think that the way it works is that the
tracking code reports visitor statistics to whatever account it's set up
under.  I'm not sure what happens when you put the code in at a url that
isn't the url of the account.  I would think that it will just track
whatever traffic the pages with the code gets, regardless of URL.

How are you going to deploy preservationarts.net in Forge and have it show
whatever is in the preservationarts.corbettresearchgroup.com site?

**Comment by meetinghouse** [comment]
I am trying to navigate Google Analytics myself, and I'm not sure how this will work. I can't really test it unless I have both URLs deployed.

The launch of lives sites is still problematic and it is quite possible I have misunderstood how it works. (I don't actually like how I think it works, so I'd be happy to find I am wrong). Here is the process I have for deploying a live site with the actual URL after having populated the .corbettresearchgroup.com one (formatted more as working notes for me, sorry):

First, create a sub-domain site in corbettresearchgroup.com.
Example: steeplejacksworld.corbettresearchgroup.com 

When ready to go live, create a new site (through Forge) with the “real” domain name.
	Example: www.steeplejacksworld.com -- be sure to include the www here.

Once the site is created, click the grey pencil icon next to it on the site list. The adding of a GitHub repository is different from test site deployment.

Steps:
	Click GitHub repository button
	 Fill out the form thus:
Provider = GitHub
Repository = alnutile/starter_forge
Branch = master
Uncheck both installation options

Now click the Install Repository button. It will take a long moment.

Now edit deploy script. Don’t click Enable Quick Deploy this time. Just click Edit Deploy script.

Delete all the default deploy script that appears.

Replace it with this line:
~/point_to.php -c www.builders.com -s builders.corbettresearchgroup.com

Edit the www.builders.com  and builders to be whatever site you are launching instead. Like www.steeplejacksworld.com and steeplejacksworld

Click Save Deploy script.

Click Deploy Now and wait until it is done spinning.

DNS set up:

Go to Bulk Register>>My Domains
Create a domain in BulkRegister for steeplejacks.com
Go to the host records and put the IP 162.243.84.17 in for the following A (Addrss) records
@ (none) and * (all others)

This creates a deployed site at steeplejacksworld.com, but it does not reflect the content changes at steeplejacksworld.corbettresearchgroup.com. I have a huge email thread with our server guy about this, which was never totally resolved, actually. The last word from him was 
"I have half the answer -- the web server isn't configured specifically
for that domain, so it goes with the default.  I don't know why that
site is the default, but that's what it is.

I manually modified the www.steeplejacksworld.com portion of the web
server configuration so it will also answer for steeplejacksworld.com.
If you enter that in your browser now you should get the correct site.

I think the configuration should be set up by Forge when the site is
deployed.  I'm going to poke around and see if I can figure it out,
but if not I'll ask Al."

**Comment by acavanagh** [comment]
^ that sounds complicated.  If your server guy is comfortable at the command line I can show him how to switch a site over to production in ~ 5minutes.  All he has to do is log in, edit one line in one file, and type one short line of code.  Same edits every time.  It gets complicated if he's not comfortable with command line management.  When you switch it over to live, do you need to keep a separate copy of the staging site, or is that not useful anymore?

**Comment by meetinghouse** [comment]
OK, I've contacted Dan Parker, the sysadmin (pasted email from him below, including his email if you need it), so please do get in touch with him to give him instructions to do this via the command line.

If you guys need an actual test site deployed in forge to test this on, please use steeplejacksworld.corbettresearchgroup.com I already launched a live version of this in the past, sort of, at steeplejacksworld.com  Feel free to delete that already-launched live site to clear the way for testing if you do decide to use it. The whole site is only for testing, but John does own the domain, so live launch can be tested with it.

From Dan: D Parker <drp@thirddeviation.com>
"I'm very comfortable at the command-line so if Andy wants to send along
the instructions I'll be your backpocket troubleshooter.  :-)  Perhaps we
can script the process to make it even easier."

**Comment by meetinghouse** [comment]
Since we have a live site now, I went ahead and set them up with Google Analytics, plugging the Tracking ID into the user interface field the was developed in this issue. It looks like it worked well -- I can see (blank) reports for the site in Google-Analytics, AND I was able to verify ownership of the site with Google Webmaster Tools by copying and pasting in the Analytics Tracking ID. Very circular, but it appears that the code went where it needed to go. Thank you!
